### DISCUSSION TITLE
Downloads often only include audio, no video

### DISCUSSION THREAD

**Discussion by MrYossu** [Tyrrrz/YoutubeExplode#683] [Q&A]
I use the following code (simplified for clarity, but no important details removed)...

```c#
YoutubeClient youtube = new();
Video video = await youtube.Videos.GetAsync(Uri);
Progress<double> progress = new();
progress.ProgressChanged += (_, d) => {
  // code removed for clarity
};
await youtube.Videos.DownloadAsync(Uri, "path\to\folder\video.mp4", o => o
  .SetContainer("mp4")
  .SetPreset(ConversionPreset.UltraFast)
  .SetFFmpegPath("path\to\ffmpeg.exe"),
  progress
  );
```

This works fine most of the time, but quite often, the downloaded file plays sound, but does not have any video.

As an example, I just tried to download `https://www.youtube.com/watch?v=kIkbGXLkc-g` and got an audio-only file. By contrast, if I try `https://www.youtube.com/watch?v=EO-cVpjEDHw`, it works fine.

Anyone any idea what I might be doing wrong? I'm using YoutubeExplode.Converter v6.2.5 in a .NET6 ASP.NET Core app.

Thanks

**Comment by Tyrrrz** [comment]
Can you debug through the code to see which streams it uses for muxing?

**Comment by MrYossu** [comment]
@Tyrrrz How do I do that with the code I posted? The reason for using this code rather than the longer code on the main page was that I just want to grab the best video.

Please can you clarify. Thanks

**Reply by Tyrrrz** [reply]
@MrYossu put a breakpoint on `youtube.Videos.DownloadAsync`, then step inside until it gets here:

https://github.com/Tyrrrz/YoutubeExplode/blob/b19acb5739de60f42f82c53b4f0a790e95f76d90/YoutubeExplode.Converter/ConversionExtensions.cs#L27-L46

Then inspect the contents of the `streamInfos` argument
